server/http/dialog/install.go

```diff
@@ -14,36 +14,38 @@ import (
 )
 
 type installDialogState struct {
-	Manifest      *apps.Manifest
+	AppID         apps.AppID
 	TeamID        string
 	LogRootPostID string
 	LogChannelID  string
 }
 
-func NewInstallAppDialog(manifest *apps.Manifest, secret, pluginURL string, commandArgs *model.CommandArgs) model.OpenDialogRequest {
+func NewInstallAppDialog(m *apps.Manifest, secret, pluginURL string, commandArgs *model.CommandArgs) model.OpenDialogRequest {
 	intro := md.Bold(
-		md.Markdownf("Application %s requires the following permissions:", manifest.DisplayName)) + "\n"
-	for _, permission := range manifest.RequestedPermissions {
+		md.Markdownf("Application %s requires the following permissions:", m.DisplayName)) + "\n"
+	for _, permission := range m.RequestedPermissions {
 		intro += md.Markdownf("- %s\n", permission.Markdown())
 	}
 	intro += md.Bold(
-		md.Markdownf("\nApplication %s requires to add the following to the Mattermost user interface:", manifest.DisplayName)) + "\n"
-	for _, l := range manifest.RequestedLocations {
+		md.Markdownf("\nApplication %s requires to add the following to the Mattermost user interface:", m.DisplayName)) + "\n"
+	for _, l := range m.RequestedLocations {
 		intro += md.Markdownf("- %s\n", l.Markdown())
 	}
 	intro += "\n---\n"
 
-	elements := []model.DialogElement{
-		{
+	elements := []model.DialogElement{}
+	if m.Type == apps.AppTypeHTTP {
```

Comment:
TODO: Talk about offline with @levb if we need that

Reply:
@hanzei did we discuss it to your satisfaction?

Reply:
Yes, resolved